### PR TITLE
Fix When serialization fails, the code value in org.apache.dubbo.rpc.RpcException is set incorrectly. (#12279)

### PR DESCRIPTION
**Pull request by xuziyang**
fix https://github.com/apache/dubbo/issues/12279

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
Can SerializationException just a type of RpcException

**Comment by xuziyang** [comment]
> RpcException

RpcException uses the internal code variable to indicate which type of exception has occurred. When a serialization exception occurs, the code value is set to NETWORK_EXCEPTION, which is incorrect. The correct value for code should be SERIALIZATION_EXCEPTION.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java [hunk 1]
```diff
@@ -203,7 +204,9 @@ private void doReceived(Response res) {
             this.complete(res.getResult());
         } else if (res.getStatus() == Response.CLIENT_TIMEOUT || res.getStatus() == Response.SERVER_TIMEOUT) {
             this.completeExceptionally(new TimeoutException(res.getStatus() == Response.SERVER_TIMEOUT, channel, res.getErrorMessage()));
-        } else {
+        } else if(res.getStatus() == Response.SERIALIZATION_ERROR){
+            this.completeExceptionally(new SerializationException(channel, res.getErrorMessage()));
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java:208]: `new RpcException(SERIALIZATION_EXCEPTION, res.getErrorMessaeg())` would be enough?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java (hunk 1 above)
Comment (xuziyang) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java:208]: As shown in the figure below, in the invoke method of AsyncToSyncInvoker, other exceptions are uniformly converted to RpcException. If you directly create a new RpcException(SERIALIZATION_EXCEPTION, res.getErrorMessage()) and throw it at the place you mentioned, would it conflict with the existing code logic?

Also, why did building and testing fail without any error messages?

image

**Review by AlbumenJ — APPROVED** [review]
LGTL

**Comment by AlbumenJ** [comment]
@CrazyHZM @chickenlj PTAL

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**